packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -118,6 +129,17 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
         return;
       }
 
+      if (manager.isLink(key)) {
+        if (linkBehavior === 'selection') {
+          router.open(ref.current, e);
+          // Always set selected keys back to what they were so that select and combobox close.
+          manager.setSelectedKeys(manager.selectedKeys);
```

Comment:
Not totally sure about this because it'll also result in onSelectionChange being fired...